llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -12252,25 +12254,47 @@ InstructionCost BoUpSLP::getSpillCost() const {
       };
 
       // Debug information does not impact spill cost.
-      if (isa<CallBase>(&*PrevInstIt) && !NoCallIntrinsic(&*PrevInstIt) &&
-          &*PrevInstIt != PrevInst)
+      // Vectorized calls, represented as vector intrinsics, do not impact spill
+      // cost.
+      if (const auto *CB = dyn_cast<CallBase>(&*PrevInstIt);
+          CB && !NoCallIntrinsic(CB) &&
+          (!isVectorized(CB) ||
+           any_of(getTreeEntries(CB), [&](const TreeEntry *TE) {
```

Comment:
Can you move this bit to a new review?  I agree we should explore in this direction, but these needs to happen once per tree entry, not once per instruction corresponding to that tree entry.  I'd also like to get your previous change - with just the isVectorized change - landed so that we can iterate on top of that.